R/event_sources.R

```diff
@@ -0,0 +1,272 @@
+  keep_source_vars = exprs(RANDDT)
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+cb_y <- event(
+  description = paste(
+    "Define CR, PR, SD, or NON-CR/NON-PD occuring at least 42 days after",
+    "randomization as clinical benefit"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("CR", "PR", "SD", "NON-CR/NON-PD") &
+    ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "Y")
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_cr <- event(
+  description = "Define complete response (CR) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "CR",
+  set_values_to = exprs(
+    AVALC = "CR"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_pr <- event(
+  description = "Define partial response (PR) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "PR",
+  set_values_to = exprs(
+    AVALC = "PR"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_sd <- event(
+  description = paste(
+    "Define stable disease (SD) for best overall respone (BOR) as CR, PR, or SD",
+    "occurring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("CR", "PR", "SD") & ADT >= RANDDT + 42,
+  set_values_to = exprs(
+    AVALC = "SD"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_non_crpd <- event(
+  description = paste(
+    "Define NON-CR/NON-PD for best overall response (BOR) as NON-CR/NON-PD",
+    "occuring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC == "NON-CR/NON-PD" & ADT >= RANDDT + 42,
+  set_values_to = exprs(
+    AVALC = "NON-CR/NON-PD"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_pd <- event(
+  description = "Define progressive disease (PD) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "PD",
+  set_values_to = exprs(
+    AVALC = "PD"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+bor_ne <- event(
+  description = paste(
+    "Define not evaluable (NE) for best overall response (BOR) as SD, NON-CR/NON-PD,",
+    "or NE (should be specified after bor_sd and bor_non_crpd)"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC %in% c("SD", "NON-CR/NON-PD", "NE"),
+  set_values_to = exprs(
+    AVALC = "NE"
+  )
+)
+
+#' @format NULL
+#' @keywords source_specifications
+#' @rdname event_objects
+#' @export
+no_data_missing <- event(
+  description = paste(
+    "Define missing response (MISSING) for all patients in adsl (should be used",
+    "as last event)"
+  ),
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(
+    AVALC = "MISSING"
```

Comment:
I have updated the text as suggested. You could use `AVALC = "NE"` instead of `AVALC = "MISSING"`. I used `AVALC = "MISSING"` because that's used in the basic version and was used in the old vignette.